_LowLevelCode/cpp/sort_pixels_by_bins/sort_pixels_by_bins.cpp

```diff
@@ -2,430 +2,397 @@
-  */
-
-  mxClassID category;
-  bool array_type_is_known(false);
-
-  category = mxGetClassID(pix_cell_array_ptr);
-  if (category != mxCELL_CLASS)
-    return "Input pixel array has to be packed in cell array";
-
-  n_tot_pixels = 0;
-
-  const mxArray *cell_element_ptr;
-
-  size_t num_of_cells = mxGetNumberOfElements(pix_cell_array_ptr);
-  pix_block_sizes.assign(num_of_cells, 0);
-  pPix_blocks.assign(num_of_cells, NULL);
-
-  /* Each cell mxArray contains 1-by-n cells; Each of these cells
-  is an 9xNpix mxArray. */
-  for (int ind = 0; ind < num_of_cells; ind++) {
-
-    cell_element_ptr = mxGetCell(pix_cell_array_ptr, ind);
-    if (cell_element_ptr != NULL) {
-      // check if a contributing pixels have the same parameters
-      category = mxGetClassID(cell_element_ptr);
-      if (category == mxDOUBLE_CLASS) {
-        if (array_type_is_known) {
-          if (single_precision)
-            return "Input pixels array contains blocks with different type of "
-                   "pixels. Only one type of pixels (single or double) is "
-                   "supported";
-        } else {
-          single_precision = false;
-        }
-      } else if (category == mxSINGLE_CLASS) {
-        if (array_type_is_known) {
-          if (!single_precision)
-            return "Input pixels array contains blocks with different type of "
-                   "pixels. Only one type of pixels (single or double) is "
-                   "supported";
-        } else {
-          single_precision = true;
+std::string  verify_pix_array(const mxArray * pix_cell_array_ptr, bool &single_precision, std::vector<size_t> &pix_block_sizes,
+    std::vector<const double *> &pPix_blocks, size_t &n_tot_pixels) {
+    /* function processes and validates cell array of input pixels
+
+    in particular, it calculates each cell size and number of pixels, containing in each array.
+    */
+
+    mxClassID   category;
+    bool array_type_is_known(false);
+
+    category = mxGetClassID(pix_cell_array_ptr);
+    if (category != mxCELL_CLASS)return "Input pixel array has to be packed in cell array";
+
+    n_tot_pixels = 0;
+
+
+    const mxArray *cell_element_ptr;
+
+    size_t num_of_cells = mxGetNumberOfElements(pix_cell_array_ptr);
+    pix_block_sizes.assign(num_of_cells, 0);
+    pPix_blocks.assign(num_of_cells, NULL);
+
+    /* Each cell mxArray contains 1-by-n cells; Each of these cells
+    is an 9xNpix mxArray. */
+    for (int ind = 0; ind < num_of_cells; ind++) {
+
+        cell_element_ptr = mxGetCell(pix_cell_array_ptr, ind);
+        if (cell_element_ptr != NULL) {
+            // check if a contributing pixels have the same parameters
+            category = mxGetClassID(cell_element_ptr);
+            if (category == mxDOUBLE_CLASS) {
+                if (array_type_is_known) {
+                    if (single_precision)return "Input pixels array contains blocks with different type of pixels. Only one type of pixels (single or double) is supported";
```

Comment:
I don't think that change of formatting has improved the layout of this code

Reply:
I reverted this file back to its original state in https://github.com/pace-neutrons/Horace/pull/495/commits/e782ba45d76a7b8861e273e545cfe5d9eca0a770